src/arrayMethodJoin.js

```diff
@@ -4,9 +4,26 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    if (this.length < 1) {
+      return '';
+    }
+
+    let str = '';
+
+    for (let i = 0; i < this.length - 1; ++i) {
+      if ((this[i]) === null || (this[i]) === undefined) {
+        str += separator;
+        continue;
+      }
+
+      str += '' + this[i] + separator;
```

Comment:
```suggestion
      str += this[i] + separator;
```
do you need an empty string here?

Reply:
> do you need an empty string here?

Yes, without it doesnt work 100%

Reply:
then you do something wrong. You could remove it.